lib/odometry-ng/include/odometer/OdometerDifferential.hpp

```diff
@@ -0,0 +1,93 @@
+#pragma once
+
+#include <cstdint>
+
+#include "encoder/EncoderInterface.hpp"
+#include "OdometerInterface.hpp"
+#include "OdometerDifferentialParameters.hpp"
+
+namespace cogip {
+
+namespace odometer {
+
+class OdometerDifferential: public OdometerInterface {
+
+public:
+    /// @brief Construct a new differential odometer object
+    /// @param parameters Odometer parameters
+    explicit OdometerDifferential(OdometerDifferentialParameters &parameters) : parameters_(parameters), left_encoder_(NULL), right_encoder_(NULL) {}
+
+    /// @brief Set the default odometry position
+    ///
+    /// @note this function should be called to reset robot position and defined a new default one
+    ///
+    /// @param x X coordinate (mm)
+    /// @param y Y coordinate (mm)
+    /// @param O angle (deg)
+    void set_position(double x, double y, double O) override
+    {
+        pos_.x = x;
+        pos_.y = y;
+        pos_.O = DEG2RAD(O);
+    }
+    
+
+    /// @brief Set the default odometry position
+    ///
+    /// @note this function should be called to reset robot position and defined a new default one
+    ///
+    /// @param pos postion reference
+    void set_position(const cogip::cogip_defs::Pose &pos) override
```

Comment:
Should we move to CamelCase ? :thinking:

Reply:
On the other class like in motion control, methods use snake case, I think we can keep it like this.